Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/border.h

```diff
@@ -80,27 +80,45 @@ namespace Polygon_mesh_processing {
 
     template<typename PM
            , typename FaceRange
-           , typename HalfedgeOutputIterator>
+           , typename HalfedgeOutputIterator
+           , typename NamedParameters>
     HalfedgeOutputIterator border_halfedges_impl(const FaceRange& faces
                                                , typename boost::cgal_no_property::type
                                                , HalfedgeOutputIterator out
-                                               , const PM& pmesh)
+                                               , const PM& pmesh
+                                               , const NamedParameters& /* np */)
     {
       return border_halfedges_impl(faces, out, pmesh);
     }
 
     template<typename PM
            , typename FaceRange
            , typename FaceIndexMap
-           , typename HalfedgeOutputIterator>
+           , typename HalfedgeOutputIterator
+           , typename NamedParameters>
     HalfedgeOutputIterator border_halfedges_impl(const FaceRange& faces
                                                , const FaceIndexMap& fmap
                                                , HalfedgeOutputIterator out
-                                               , const PM& pmesh)
+                                               , const PM& pmesh
+                                               , const NamedParameters& /* np */)
     {
       typedef typename boost::graph_traits<PM>::halfedge_descriptor halfedge_descriptor;
       typedef typename boost::graph_traits<PM>::face_descriptor     face_descriptor;
 
+      //make a minimal check that it's properly initialized :
+      //if the 2 first faces have the same id, we know the property map is not initialized
+      if (boost::is_same<typename GetFaceIndexMap<PM, NamedParameters>::Is_internal_map,
+                         boost::true_type>::value)
+      {
+        typename boost::range_iterator<const FaceRange>::type it = boost::const_begin(faces);
+        if (get(fmap, *it) == get(fmap, *cpp11::next(it)))
```

Comment:
There are already two checks in the main function `border_halfedges()`:
` if (faces.empty()) return out;`
and
` if (boost::is_same<FIMap, Unset_FIMap>::value || faces.size() == 1)`
who should handle those cases. Do I repeat them there ?